lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -590,6 +594,35 @@ withNetworkLayer tr network np project k = do
             percentage = currentSlot % latestSlot
         pure $ Syncing $ Quantity $ clipToPercentage percentage
 
+    stakePoolsSummary :: BF.ClientConfig -> Coin -> IO StakePoolsSummary
+    stakePoolsSummary bfConfig coin = do
+        protocolParameters <- currentProtocolParameters bfConfig
+        runBFM bfConfig do
+            BF.Network{_networkStake = BF.NetworkStake{_stakeLive}} <-
+                BF.getNetworkInfo
+            totalLiveStake <- fromBlockfrostM _stakeLive
+            pools <- mapConcurrently BF.getPool =<< BF.listPools
+            stake <- poolsStake totalLiveStake pools
+            pure StakePoolsSummary
+                { nOpt = intCast $ desiredNumberOfStakePools protocolParameters
+                , rewards = Map.empty
+                , stake
+                }
+          where
+            poolsStake
+                :: MonadError BlockfrostError m
+                => Coin
+                -> [BF.PoolInfo]
+                -> m (Map PoolId Percentage)
+            poolsStake total = fmap Map.fromList . traverse \BF.PoolInfo{..} ->
+                (,) <$> fromBlockfrostM _poolInfoPoolId <*> do
+                    live <- fromBlockfrostM @_ @Coin _poolInfoLiveStake
+                    let ratio = Coin.toInteger live % Coin.toInteger total
+                    case mkPercentage ratio of
```

Comment:
(Not a review comment, no need to change.)

By the way, I started to find functions helpful that are total, but "somewhat wrong" — e.g. `clipToPercentage`. Partial functions in Haskell are a thorny issue — but I found the paper you mentioned, [Noonan, "Ghosts of departed proofs" (2018)](https://kataskeue.com/gdp.pdf) very insightful, section 1.1. was extremely lucid. I hope that we can revisit this in the Reading Club at some point.